### PR TITLE
added tests for cucumber, improved PICA, modified view to ignore miss…

### PR DESCRIPTION
**Pull request by heqiu**
…ing data

### CODE REVIEW

**Comment by heqiu** [comment]
Note: this causes Rspec to fail.  We need to merge the code with Manoj & Shalini's implementation of dealing with missing data for rspecs to pass.

File: db/schema.rb [hunk 1]
```diff
@@ -13,13 +13,6 @@
 
 ActiveRecord::Schema.define(version: 20160418150933) do
 
-  create_table "course_names", force: true do |t|
```
Comment (brentwalther) [inline db/schema.rb:16]: You'll need to make sure and merge these changes in from master. This part shouldn't be in the diff.

File: app/models/evaluation.rb [hunk 2]
```diff
@@ -43,4 +43,6 @@ def self.default_sorted_groups
   def is_honors_section?
     section.to_s.starts_with?("2")
   end
+
+  scope :with_enrollment_data, -> {where.not(enrollment: nil)}
```
Comment (brentwalther) [inline app/models/evaluation.rb]: While this scope achieves what you want, you should probably use the one that manoj and shalini came up with in `/pull/10`

File: app/controllers/evaluation_controller.rb [hunk 3]
```diff
@@ -107,9 +112,12 @@ def upload_gpr
 
       flash[:notice] = "#{num_new_records} new GPRs imported. #{num_updated_records} evaluation GPRs updated."
       redirect_to evaluation_index_path
-    else
+    elsif params[:data_file] != nil
```
Comment (brentwalther) [inline app/controllers/evaluation_controller.rb:119]: Is there a specific reason this was moved?

**Comment by brentwalther** [comment]
Looks good, please adjust the scope to use the one manoj and shalini created.

File: app/controllers/evaluation_controller.rb (hunk 3 above)
Comment (heqiu) [inline app/controllers/evaluation_controller.rb:119]: Not sure what you mean; we added handling the case of uploading before selecting a file in both pica upload and GPR upload, which caused line numbers to shift

Comment (brentwalther) [inline app/controllers/evaluation_controller.rb:119]: Ah, alright. Never mind

**Pull request merged**